### ISSUE TITLE
'MADDPGAgentTrainer' object has no attribute 'actor_target'

### ISSUE THREAD

**Issue by msclar** [hebowei2000/maddpg_tf2#1]
Hi!

When I try to run your code, I get the same error that you show in the notebook. Even if I remove the commented lines in maddpg.py, I am not able to run it sucessfully.

Is it possible that the repo is missing a commit?

Thank you for porting MADDPG to TF2!

**Comment by benalcazardiego** [comment]
Hello!

Same problem here. Could you solve it? I saw it I first called in line 194 in maddpg.py, yet it is not defined anywhere.

Thanks

**Comment by jonmenard** [comment]
I am struggling with the same thing, does anyone have a solution?

**Comment by Nikhil-ostwal** [comment]
Any updates yet? I am also struggling to implement MADDPG in TF2.

**Comment by jonmenard** [comment]
I have it working now. The trainer can be adapted for your scenario and should be pretty easy to implement. I had to change a few lines in the maddpy.py files to get it working. Try something like this: 

- uncomment lines 85 and 86
- change line 229 to: state_action_n = tf.concat((obs_n, act_n), axis=-1)
- change line 230 to: state_action_attached = tf.reshape(state_action_n, [batch_size, self.n * (self.act_size+self.obs_size)])

** you may also need to reshape some lists around lines 170 - 174. For example, line 172 I have "obs_n.append(obs.reshape(FLAGS.batch_size, self.obs_size))" however this is because of how I store my lists in the replay memory so it may be different for you.

I would share my code, however it is currently confidential with my school

**Comment by Nikhil-ostwal** [comment]
> I have it working now. The trainer can be adapted for your scenario and should be pretty easy to implement. I had to change a few lines in the maddpy.py files to get it working. Try something like this:
> 
> * uncomment lines 85 and 86
> * change line 229 to: state_action_n = tf.concat((obs_n, act_n), axis=-1)
> * change line 230 to: state_action_attached = tf.reshape(state_action_n, [batch_size, self.n * (self.act_size+self.obs_size)])
> 
> ** you may also need to reshape some lists around lines 170 - 174. For example, line 172 I have "obs_n.append(obs.reshape(FLAGS.batch_size, self.obs_size))" however this is because of how I store my lists in the replay memory so it may be different for you.
> 
> I would share my code, however it is currently confidential with my school


Thank you for the explanation, that was really helpful. I have a few questions :

-  Do I need to write the method _mlp_model_  in line 51 in _train.py_
-  In maddpg.py added this line : actor.compile(loss = "mse", optimizer = Adam(lr = self.learning_rate, clipnorm=0.5))  in _build_model_ . Is it correct?
- Also I get an error while reshaping in line 172 in maddpg.py that 'MADDPGAgentTrainer' object has no attribute 'batch_size'
